### PR TITLE
change spring/damping calculation, added centre of mass option

### PR DESCRIPTION
**Pull request by michurebmann**
The spring/damper value will now be calculatet based on absolut values, means when you change the spring length, the spring/damp stiffnes will be the same.

when u work with normalized values, the spring/damp stiffness will change when you change the spring length

added a node "CentreOfMass".
it will change the com, based on the nodes position, so you can change the COM without changing the collider size/pos. could manipulate it to represent the fuel consume

### CODE REVIEW

**Comment by michurebmann** [comment]
I hope that's how you do it.

**Comment by Dechode** [comment]
Hey! Thanks for the pull request. For the most part this seems fine, but ever since Godot 4 it has been possible to set center of gravity of a RigidBody from the inspector, so there is no need for a node to set it. One just needs to set the center_of_mass_mode to custom and you can then specify a Vector3 that will be used as center of mass.  I'll also leave few comments on naming conventions for variables.

**Comment by michurebmann** [comment]
ok, i did nor know that with the com option... lol.

as far as i know i can change my pull request by change the sugested inputs from you and pull again?

i never done this, where can i see the comments on the naming conventions?

**Comment by Dechode** [comment]
> as far as i know i can change my pull request by change the sugested inputs from you and pull again?
Yeah thats how i imagine it works, i am actually quite new to collaborating on github myself. I have some experience with gitlab and thats how it was done there. 

The comments should be visible somewhere in this thread and / or in the files changed tab. You were so fast to reply i probably had not written them yet lol

**Comment by Dechode** [comment]
The dot net property is fine, i can still open the project with non .Net version of Godot. 
Weird that the comments are not showing up. Not even in this thread? I see a section saying "Dechode started a review" and that's where the comments are. I left some comments again in the newest change about typos in comments in code.
Other than that it looks good now, I will try out the changes soon and then merge this after those few typos are corrected.

File: scenes/vehicle/wheel_suspension.gd [hunk 1]
```diff
@@ -11,10 +11,10 @@ extends RayCast3D
 @export var rebound = 4.0
 @export var anti_roll = 0.0
 
-var mm_load_spring:float = 0
-var prev_mm_load_spring:float = 0
-var mm_per_second:float = 0
-var N_load_spring:float = 0
+var spring_load_mm:float = 0
+var prev_spring_load_mm:float = 0
+var spring_speed_mm_per_seconds:float = 0
```
Comment (michurebmann) [inline scenes/vehicle/wheel_suspension.gd:16]: maybe too long name?

Comment (Dechode) [reply scenes/vehicle/wheel_suspension.gd:16]: Hmm, i think long variable names are fine, if they are descriptive. Which this is IMO. I think it is fine

**Comment by Dechode** [comment]
I can see your comment about variable name maybe being too long. I try to request a review from myself again, maybe that has something to do with my comments not showing up.

**Comment by michurebmann** [comment]
i see the review request, but thats for you, i cannot open it

File: scenes/vehicle/base_car.tscn [hunk 2]
```diff
@@ -1,16 +1,11 @@
-[gd_scene load_steps=9 format=3 uid="uid://exyybiakl0yk"]
+[gd_scene load_steps=7 format=3 uid="uid://exyybiakl0yk"]
 
 [ext_resource type="Script" path="res://scenes/vehicle/base_car.gd" id="1_n8n57"]
-[ext_resource type="AudioStream" path="res://sounds/tres/engine_sample.tres" id="2_om5dm"]
```
Comment (Dechode) [inline scenes/vehicle/base_car.tscn:4]: Did the AudioStream get deleted somehow?

File: scenes/vehicle/base_car.tscn [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
 [ext_resource type="Script" path="res://scenes/camera.gd" id="3_5upga"]
 [ext_resource type="Script" path="res://scenes/vehicle/wheel_suspension.gd" id="4_6hvcn"]
 [ext_resource type="Material" path="res://materials/tire.tres" id="5_cuc4d"]
 [ext_resource type="PackedScene" uid="uid://djbt3ltge1aat" path="res://scenes/gui/gui.tscn" id="6_cp0hb"]
 
-[sub_resource type="Curve" id="8"]
-_data = [Vector2(0, 0.563636), 0.0, 0.0, 0, 0, Vector2(0.575472, 1), 0.0, 0.0, 0, 0, Vector2(0.962264, 0.809091), -0.335178, -0.335178, 0, 0, Vector2(1, 0.109091), -6.85253, 0.0, 0, 0]
-point_count = 4
-
```
Comment (Dechode) [inline scenes/vehicle/base_car.tscn:13]: Did these get deleted as well? Or just github being weird

File: scenes/vehicle/base_car.tscn [hunk 4]
```diff
@@ -23,14 +18,6 @@ transform = Transform3D(1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0.5, 0)
...
 mass = 700.0
 script = ExtResource("1_n8n57")
-max_brake_force = 5000.0
-rpm_clutch_out = 2000.0
-torque_curve = SubResource("8")
-engine_moment = 0.2
-engine_sound = ExtResource("2_om5dm")
-gear_ratios = [3.5, 3.0, 2.5, 1.72, 1.2, 1.0]
-gear_inertia = 0.09
-rear_diff = 2
```
Comment (Dechode) [inline scenes/vehicle/base_car.tscn:33]: And did something happen to these as well?

**Review by Dechode — CHANGES_REQUESTED** [review]
Okay maybe you can see all the comments now?

**Comment by michurebmann** [comment]
now i can see it all!  

well, its weird, my project works just fine,
i think its because i first added a node (com) which mix up all the indexes of the scene.

i will make a complete new pull request, i think it will be better.

can you just reject the pull? i will then re synk the fork and redo the pull, since there are now a lot less changes.
or ho can i undo the pull?

**Pull request closed without merging**